sdk/main/include/Defaults.h

```diff
@@ -38,6 +38,10 @@ constexpr auto DEFAULT_MIN_BACKOFF = std::chrono::milliseconds(250);
  * The default maximum duration of time to wait before retrying to submit a previously-failed request.
  */
 constexpr auto DEFAULT_MAX_BACKOFF = std::chrono::seconds(8);
+/**
+ * The default maximum amount of time to spend on one execution attempt
+ */
+constexpr auto DEFAULT_GRPC_DEADLINE = std::chrono::seconds(10);
```

Comment:
Just curious. Is this by specification? Can we pick a better value or have some kind of logic regarding the amount of time spent?

Reply:
Not 100% sure, just stole this from the other SDKs. I'm assuming there was some discussion at some point about this or there was some testing done to determine this being the best value. @SimiHunjan may be able to provide more insight